src/query/functions/src/srfs/mod.rs

```diff
@@ -278,3 +329,49 @@ fn build_unnest(
         }),
     }
 }
+
+fn unnest_variant_array(
+    val: &[u8],
+    row: usize,
+    max_nums_per_row: &mut [usize],
+) -> (Value<AnyType>, usize) {
+    let mut len = 0;
+    let mut builder = StringColumnBuilder::with_capacity(0, 0);
```

Comment:
We only need a `builder` if the `array_values` function returns some value other than `None`, otherwise we can just return `Scalar::Null`.